cumulus/parachains/runtimes/assets/asset-hub-westend/src/genesis_config_presets.rs

```diff
@@ -0,0 +1,170 @@
+};
+
+const ASSET_HUB_WESTEND_ED: Balance = ExistentialDeposit::get();
+
+fn asset_hub_westend_genesis(
+	invulnerables: Vec<(AccountId, AuraId)>,
+	endowed_accounts: Vec<AccountId>,
+	endowment: Balance,
+	id: ParaId,
+) -> serde_json::Value {
+	let config = RuntimeGenesisConfig {
+		balances: BalancesConfig {
+			balances: endowed_accounts.iter().cloned().map(|k| (k, endowment)).collect(),
+		},
+		parachain_info: ParachainInfoConfig { parachain_id: id, ..Default::default() },
+		collator_selection: CollatorSelectionConfig {
+			invulnerables: invulnerables.iter().cloned().map(|(acc, _)| acc).collect(),
+			candidacy_bond: ASSET_HUB_WESTEND_ED * 16,
+			..Default::default()
+		},
+		session: SessionConfig {
+			keys: invulnerables
+				.into_iter()
+				.map(|(acc, aura)| {
+					(
+						acc.clone(),          // account id
+						acc,                  // validator id
+						SessionKeys { aura }, // session keys
+					)
+				})
+				.collect(),
+			..Default::default()
+		},
+		polkadot_xcm: PolkadotXcmConfig {
+			safe_xcm_version: Some(SAFE_XCM_VERSION),
+			..Default::default()
+		},
+		..Default::default()
+	};
+
+	serde_json::to_value(config).expect("Could not build genesis config.")
+}
+
+/// Encapsulates names of predefined presets.
+mod preset_names {
+	pub const PRESET_LOCAL: &str = "local";
+	pub const PRESET_GENESIS: &str = "genesis";
+}
+
+/// Provides the JSON representation of predefined genesis config for given `id`.
+pub fn get_preset(id: &PresetId) -> Option<Vec<u8>> {
+	use preset_names::*;
+	let patch = match id.try_into() {
+		Ok(PRESET_GENESIS) => asset_hub_westend_genesis(
+			// initial collators.
+			vec![
+				(
+					hex!("9cfd429fa002114f33c1d3e211501d62830c9868228eb3b4b8ae15a83de04325").into(),
+					hex!("9cfd429fa002114f33c1d3e211501d62830c9868228eb3b4b8ae15a83de04325")
+						.unchecked_into(),
+				),
+				(
+					hex!("12a03fb4e7bda6c9a07ec0a11d03c24746943e054ff0bb04938970104c783876").into(),
+					hex!("12a03fb4e7bda6c9a07ec0a11d03c24746943e054ff0bb04938970104c783876")
+						.unchecked_into(),
+				),
+				(
+					hex!("1256436307dfde969324e95b8c62cb9101f520a39435e6af0f7ac07b34e1931f").into(),
+					hex!("1256436307dfde969324e95b8c62cb9101f520a39435e6af0f7ac07b34e1931f")
+						.unchecked_into(),
+				),
+				(
+					hex!("98102b7bca3f070f9aa19f58feed2c0a4e107d203396028ec17a47e1ed80e322").into(),
+					hex!("98102b7bca3f070f9aa19f58feed2c0a4e107d203396028ec17a47e1ed80e322")
+						.unchecked_into(),
+				),
+			],
+			Vec::new(),
+			ASSET_HUB_WESTEND_ED * 4096,
+			1000.into(),
+		),
+		Ok(PRESET_LOCAL) => asset_hub_westend_genesis(
+			// initial collators.
+			vec![
+				(
+					get_account_id_from_seed::<sr25519::Public>("Alice"),
+					get_collator_keys_from_seed::<AuraId>("Alice"),
+				),
+				(
+					get_account_id_from_seed::<sr25519::Public>("Bob"),
+					get_collator_keys_from_seed::<AuraId>("Bob"),
+				),
+			],
+			vec![
+				get_account_id_from_seed::<sr25519::Public>("Alice"),
```

Comment:
`Keyring::iter()` would give the same, but I donno why no one uses it 🤷

Reply:
Maybe exactly this set of keys was required / intended? And if `Keyring` ever gets extended, new keys would also be included here?
I think iter is fine, but also not aware of historical background.

Reply:
@kianenigma I've added your comment here: https://github.com/paritytech/polkadot-sdk/issues/5705